Spring-2019/linked_lists/SOLUTIONS_2019Spring.md

```diff
@@ -89,24 +89,129 @@ If the problem is simple enough, remove this section. TODO :bug:
 
 TODO :bug:
 
-## Solutions
+Source: TODO :bug:
+
+#### Scenario
+
+Problem Statement TODO :bug:
 
-### 1. PROBLEM 1 TODO :bug:
+#### Example Input
+
+If the problem is simple enough, remove this section. TODO :bug:
+
+#### Function Signature
+
+TODO :bug:
+
+### 3. PROBLEM 2 TODO :bug:
 
 Source: TODO :bug:
 
-#### Naive/Simple Solution
+#### Scenario
+
+Problem Statement TODO :bug:
+
+#### Example Input
+
+If the problem is simple enough, remove this section. TODO :bug:
+
+#### Function Signature
 
 TODO :bug:
 
+
+## Solutions
+
+### 1. Reverse Nodes in k-Group
+
+Source: [GreeksforGreeks](https://www.geeksforgeeks.org/reverse-a-list-in-groups-of-given-size/)
+
 #### Optimal Solution
 
-TODO :bug:
+The optimal solution for this problem is to reverse the first sub-list of size `k` while keeping track of the next node and previous node. Then, recursively call for rest of the list and link the two sub-lists toghether. 
+
+Time Complexity: `O(n)` where n is the number of nodes in the given list.
+
+Here is a solution method:
+
+C++:
 
+```c++
+/* Reverses the linked list in groups 
+of size k and returns the pointer 
+to the new head node. */
+ListNode* reverseKGroup(ListNode* head, int k) {
+  ListNode* current = head;  
+  ListNode* next = NULL;  
+  ListNode* prev = NULL;  
+  int count = 0;  
+
+  //reverse first k nodes of the linked list 
+    while(current != NULL && count < k) {  
+       next = current->next;  
+       current->next = prev;  
+       prev = current;  
+       current = next;  
+       count++;  
+   }  
+ /* next is now a pointer to (k+1)th node  
```

Comment:
indentation starts getting a bit funky here, but it isn't `python` so it's not a huge deal